src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -391,19 +394,33 @@ private async Task<long> InsertGlobalStateTableRowAsync(SqlConnection connection
 
             string insertRowGlobalStateTableQuery = $@"
                 {AppLockStatements}
-
+                
                 IF NOT EXISTS (
                     SELECT * FROM {GlobalStateTableName}
                     WHERE UserFunctionID = '{this._userFunctionId}' AND UserTableID = {userTableId}
                 )
+                BEGIN
```

Comment:
Could you please add a comment explaining why do we need _oldUserFunctionId and whats is GlobalStateTableName

Reply:
Yes, this for back compatibility. We have the version at which the function is in the GlobalState table, this is to copy that version from the `oldUserFunctionId` to the new `userFunctionId` and continue the function.